src/m365/spo/commands/spo-search.ts

```diff
@@ -313,9 +331,32 @@ class SpoSearchCommand extends SpoCommand {
   }
 
   private getSelectPropertiesArray(args: CommandArgs): string[] {
-    return args.options.selectProperties
+    const selectProperties = args.options.selectProperties
       ? args.options.selectProperties.split(",")
       : ["Title", "OriginalPath"];
+
+    if (args.options.allResults) {
+      selectProperties.push("DocId");
+    }
+
+    return selectProperties;
+  }
+
+  private getSortList(args: CommandArgs): string {
+    const sortList = [];
+    if (args.options.allResults) {
+      sortList.push(formatting.encodeQueryParameter(`[DocId]:ascending`));
```

Comment:
If we put this as the first sorting criterium, won't this lead to different results than what the user requested?

Reply:
It might, yes. And yet that's what Microsoft advises...odd though, right?
We should maybe also implement a search command that uses Microsoft Graph. That does not have the same issue.